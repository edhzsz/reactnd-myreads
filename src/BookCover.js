import React from 'react'

const BookCover = ({ imageLinks }) => {
  const url = imageLinks && (imageLinks.thumbnail || imageLinks.smallThumbnail);

  const style = url
    ? { width: 128, height: 192, backgroundImage: `url("${url}")` }
    : { width: 128, height: 192 };
    
  return (
    <div className="book-cover" style={style}></div>
  );
}

export default BookCover