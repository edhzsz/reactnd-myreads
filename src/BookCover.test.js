import React from 'react';
import ReactDOM from 'react-dom';
import BookCover from './BookCover.js';
import renderer from 'react-test-renderer';

describe('Component: BookCover', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BookCover />, div);
  });

  it('should match its empty snapshot', () => {
    const tree = renderer.create(
      <BookCover />
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render using the thumbnail url as background', () => {
    const imageLinks = { thumbnail: "test-url" };
    const tree = renderer.create(
      <BookCover imageLinks={imageLinks} />
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render the smallthumbnail if thumbnail is not available', () => {
    const imageLinks = { smallThumbnail: "test-url-small" };
    const tree = renderer.create(
      <BookCover imageLinks={imageLinks} />
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render the thumbnail if both, thumbnail and smallThumbnail, are available', () => {
    const imageLinks = { thumbnail: "test-url", smallThumbnail: "test-url-small" };
    const tree = renderer.create(
      <BookCover imageLinks={imageLinks} />
     ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});