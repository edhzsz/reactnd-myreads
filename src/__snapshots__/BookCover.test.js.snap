// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: BookCover should match its empty snapshot 1`] = `
<div
  className="book-cover"
  style={
    Object {
      "height": 192,
      "width": 128,
    }
  }
/>
`;

exports[`Component: BookCover should render the smallthumbnail if thumbnail is not available 1`] = `
<div
  className="book-cover"
  style={
    Object {
      "backgroundImage": "url(\\"test-url-small\\")",
      "height": 192,
      "width": 128,
    }
  }
/>
`;

exports[`Component: BookCover should render the thumbnail if both, thumbnail and smallThumbnail, are available 1`] = `
<div
  className="book-cover"
  style={
    Object {
      "backgroundImage": "url(\\"test-url\\")",
      "height": 192,
      "width": 128,
    }
  }
/>
`;

exports[`Component: BookCover should render using the thumbnail url as background 1`] = `
<div
  className="book-cover"
  style={
    Object {
      "backgroundImage": "url(\\"test-url\\")",
      "height": 192,
      "width": 128,
    }
  }
/>
`;
